ui/public/loadConfig.js

```diff
@@ -0,0 +1,20 @@
+// Load a json config.
+
+// Run loader as an anonymous function to protect the global namespace.
+// This function accesses a file "config.json" expected to be at the
+// root of the server and loads it into the global `GlobalReactConfig`.
+(function () {
+    var xobj = new XMLHttpRequest();
+    xobj.overrideMimeType("application/json");
+    // Run this in synchronous mode since the configuration should be loaded
+    // before the React app initializes.
+    xobj.open('GET', 'config.json', false);
+    xobj.onreadystatechange = function () {
+        if (xobj.readyState == 4 && xobj.status == "200") {
```

Comment:
hmm.. won't this be called with the other readystates before this? (thereby triggering your console.log statement?). In the JMUI we added callbacks on `onload()` and `onerror()` here (I haven't dug too deploy as to whether that's 100% the right thing to do).

Reply:
good catch. I solved this a different way that now ensures that the config is loaded before loading config dependent parts of the site. See src/components/ConfigLoader.js